### PR TITLE
Do not raise when accessing the metadata of editable packages

### LINKED ISSUE #5424: conan remote remove fails with editable packages
Trying to update the metadata, it fails because the `PackageEditableLayout` has no `update_metadata()`. The main questions are:
- Should the editable packages still have the metadata methods available? 
or 
- Should we force always to retrieve the `PackageCacheLayout` sometimes, e.g from the `remote_registry.py`?

```
Traceback (most recent call last):                                                                                                                                                                                                                                             
  File "/home/luism/workspace/conan_sources/conans/client/command.py", line 1832, in run                                                                                                                                                                                       
    method(args[0][1:])                                                                                                                                                                                                                                                        
  File "/home/luism/workspace/conan_sources/conans/client/command.py", line 1423, in remote                                                                                                                                                                                    
    return self._conan.remote_remove(remote_name)                                                                                                                                                                                                                              
  File "/home/luism/workspace/conan_sources/conans/client/conan_api.py", line 77, in wrapper                                                                                                                                                                                   
    return f(*args, **kwargs)
  File "/home/luism/workspace/conan_sources/conans/client/conan_api.py", line 922, in remote_remove
    return self._cache.registry.remove(remote_name)
  File "/home/luism/workspace/conan_sources/conans/client/cache/remote_registry.py", line 301, in remove
    with self._cache.package_layout(ref).update_metadata() as metadata:
AttributeError: 'PackageEditableLayout' object has no attribute 'update_metadata'
```

### PR DESCRIPTION
**Pull request by danimtb**
Changelog: Bugfix: Do not raise when accessing the metadata of editable packages
Docs: omit

- [x] Refer to the issue that supports this Pull Request: closes #5424
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Comment by danimtb** [comment]
ok, now I got it. The last conclusion describing the implementation was not clear to me on what to do. Thanks for the clarification.

File: conans/test/functional/command/remote_test.py [hunk 1]
```diff
@@ -392,3 +392,35 @@ def test_invalid_url(self):
...
+        self.client.run("remote update remote2 %sfake" % self.servers["remote1"].fake_url)
+        self.client.run("remote rename remote2 remote-fake")
+        self.client.run("editable remove pkg/1.1@lasote/stable")
+        self.client.run("remote list_ref")
+        # Check associated remote has changed name
+        self.assertIn("pkg/1.1@lasote/stable: remote-fake", self.client.out)
+        # Check remove
+        self.client.run("editable add . pkg/1.1@lasote/stable")
+        self.client.run("remote remove remote-fake")
+        # Check clean
```
Comment (lasote) [inline conans/test/functional/command/remote_test.py]: assert

File: conans/test/functional/command/remote_test.py [hunk 2]
```diff
@@ -392,3 +392,35 @@ def test_invalid_url(self):
...
+        self.client.run("editable remove pkg/1.1@lasote/stable")
+        self.client.run("remove -f pkg/1.1@lasote/stable")
+        self.client.run("remote add remote1 %s" % self.servers["remote1"].fake_url)
+        self.client.run("install pkg/1.1@lasote/stable")
+        self.client.run("editable add . pkg/1.1@lasote/stable")
+        self.client.run("remote clean")
```
Comment (lasote) [inline conans/test/functional/command/remote_test.py]: assert

File: conans/client/cache/remote_registry.py [hunk 3]
```diff
@@ -301,52 +313,54 @@ def update(self, remote_name, url, verify_ssl=True, insert=None):
...
+                        if pkg_metadata.remote not in remotes:
+                            pkg_metadata.remote = None
 
-        remotes.save(self._filename)
+            remotes.save(self._filename)
 
     def rename(self, remote_name, new_remote_name):
         remotes = self.load_remotes()
         remotes.rename(remote_name, new_remote_name)
+        with self._editables_metadata_from_cache():
```
Comment (SSE4) [inline conans/client/cache/remote_registry.py]: is it possible to exract method? similar code repeats 5 times

Comment (danimtb) [reply conans/client/cache/remote_registry.py]: IMO that should be done in a refactor PR, not in this one. I don't refuse to make the changes here but I would like to wait for other reviews before. Thanks!

Comment (SSE4) [reply conans/client/cache/remote_registry.py]: okay

**Comment by jgsogo** [comment]
I would put this new contextmanager closer to the `EditablePackages::override` function.

...and I would suggest (for a different PR, once this one is done) merging both functions into a context manager that accepts an argument with the dictionary to use (the overriding one for the workspaces, and an empty one for this use case).

File: conans/client/cache/remote_registry.py [hunk 4]
```diff
@@ -267,6 +268,16 @@ def _validate_url(self, url):
         else:
             self._output.warn("The URL is empty. It must contain scheme and hostname.")
 
+    @contextmanager
+    def _editables_metadata_from_cache(self):
+        """
+        Hide editable packages to get the cache layout instead of the editable one
+        """
+        editables = self._cache.editable_packages
+        self._cache.editable_packages = {}
```
Comment (jgsogo) [inline conans/client/cache/remote_registry.py]: The member `editable_packages` is not a dictionary, it is an instance of `EditablePackages` class, we should override the `_edited_refs` member. I would like to keep the same type, one never knows if there is a `isinstance` check somewhere.

File: conans/test/functional/command/remote_test.py [hunk 5]
```diff
@@ -392,3 +392,43 @@ def test_invalid_url(self):
                       self.client.user_io.out)
         self.client.run("remote list")
         self.assertIn("pepe.org", self.client.out)
+
+    def test_metadata_editable_packages(self):
+        """
+        conan remote remove fails with editable packages
```
Comment (jgsogo) [inline conans/test/functional/command/remote_test.py]: Be positive ;D

```suggestion
        Check that 'conan remote' commands work with editable packages
```

**Pull request merged**